### PR TITLE
UI-672 Setting up a project, environment, and keys for PubNub for Android

### PR DESCRIPTION
**Pull request by Chesteer89**

### CODE REVIEW

File: buildSrc/src/main/java/com/pubnub/components/buildsrc/Dependencies.kt [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+        const val kotlin = "org.awaitility:awaitility-kotlin:$version"
+    }
+
+    object Hamcrest {
+        private const val version = "2.1"
+        const val hamcrest = "org.hamcrest:hamcrest:$version"
+    }
+
+    object Retrofit {
+        private const val version = "2.6.2"//"2.9.0"
```
Comment (bartk) [inline buildSrc/src/main/java/com/pubnub/components/buildsrc/Dependencies.kt]: commented code

File: buildSrc/src/main/java/com/pubnub/components/buildsrc/Dependencies.kt [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+        }
+
+        object Test {
+            private const val version = "1.4.1-alpha03"
+            const val runner = "androidx.test:runner:$version"
+            const val rules = "androidx.test:rules:$version"
+            const val core = "androidx.test:core-ktx:$version"
+
+            object Ext {
+                const val junit = "androidx.test.ext:junit-ktx:1.1.4-alpha03"
```
Comment (bartk) [inline buildSrc/src/main/java/com/pubnub/components/buildsrc/Dependencies.kt:123]: I think we can introduce a bit of consistency here. Version here already defined as const few lines above.
Maybe we shall extract all versions to corresponding constants?

File: build.gradle [hunk 3]
```diff
@@ -0,0 +1,64 @@
+import com.pubnub.components.buildsrc.Libs
+import com.pubnub.components.buildsrc.Urls
+import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
+
+apply plugin: "com.github.ben-manes.versions"
+
+buildscript {
+    ext {
+        compose_version = '1.1.0-beta04'
```
Comment (bartk) [inline build.gradle]: Can we take it form buildSrc versions?

Comment (bartk) [reply build.gradle]: I mean dereference a const val.

File: getting-started/build.gradle.kts [hunk 4]
```diff
@@ -0,0 +1,76 @@
+import com.pubnub.components.buildsrc.Libs
+
+plugins {
+    id("com.android.application")
+    id("kotlin-android")
+}
+
+tasks.register("wrapper", Wrapper::class){
```
Comment (bartk) [inline getting-started/build.gradle.kts]: Do you know if that can be moved to project level build.gradle?

Comment (Chesteer89) [reply getting-started/build.gradle.kts]: Looks like this task can be removed once we've got `gradle-wrapper.properties` in subproject

**Comment by Chesteer89** [comment]
Should be merged after chat components release (0.1.2)

File: buildSrc/src/main/java/com/pubnub/components/buildsrc/Dependencies.kt (hunk 2 above)
Comment (Chesteer89) [reply buildSrc/src/main/java/com/pubnub/components/buildsrc/Dependencies.kt:123]: The problem here is that androidx.test package have version `1.4.1` when extension is only available in `1.1.4`. There're many places like this one, but those versions are not reused anywhere. That's why it's not extracted to const.

Similar story with `androidx.compose` packages. I used there two consts, one for current version and one for previous - devs always forget to release all the packages with newest compose release...

**Review by bartk — APPROVED** [review]

**Review by Chesteer89 — COMMENTED** [review]
Ready to merge

**Pull request merged**